youtube_dl/extractor/rutube.py

```diff
@@ -17,7 +17,7 @@
 class RutubeIE(InfoExtractor):
     IE_NAME = 'rutube'
     IE_DESC = 'Rutube videos'
-    _VALID_URL = r'https?://rutube\.ru/(?:video|(?:play/)?embed)/(?P<id>[\da-z]{32})'
+    _VALID_URL = r'https?://rutube\.ru/(?:video|(?:play/)?embed)/(?P<id>[\da-z]{32})(/)?$'
```

Comment:
There are several reasons why this strikes me as a non-optimal solution. We're really only asserting that no previous regexes match our URL; this makes no positive case for our playlist URL, we don't test for its characteristics. Its probably not good for further functionality on that site; we don't know what else youtube-dl may want to support in the future. Adopting that approach may lead someone to add a different extractor in the future and our playlist extractor may match that specification, simply because its not defined properly. Trying to then fix the issue makes us fiddle with parts of the program that aren't concerned with the new extractor.

Besides, that only brings us back to the question of how the regular expression for two parameters in a URL with arbitrary order can be resolved. I don't see your approach solving much of a problem.